deploy/vagrant/bin/init.py

```diff
@@ -13,10 +15,19 @@
 import sys
 from sys import stderr
 import argparse
+import json
+import tempfile
+import errno
+
+import boto
+from boto import ec2
+import yaml
 
 def parse_args():
     parser = argparse.ArgumentParser()
     parser.add_argument('provider')
+    parser.add_argument('--is-aws-spot', action='store_true')
```

Comment:
Is this argument necessary anymore, since you have the new script?

If this file only does something when provider is 'aws', then it should not be called 'init.py', and it should not require the provider argument.